### ISSUE TITLE
Light color theme making it impossible to read the text without highlighting it in iTerm2.

### ISSUE THREAD

**Issue by Biodiscus** [siderolabs/talos#9490]
## Bug Report

The way the colors are handled within Talos are preventing me from using a lighter theme while using `talosctl dashboard`. 

### Description

I've forked the repo and made a fix for myself [here](https://github.com/Biodiscus/talos/commit/4a22a44a86974f885166180183934d61ef21fe8d), which uses the terminal colors instead. The reason why I have made it a fork for the time being, is because the colors become less vibrant in darker terminals so it's not the perfect solution but works for me :)

When I'm using a lighter color scheme on iTerm2 the dashboard looks like:
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/a3e5c411-4da2-483a-8341-8fa3e4e7c1ec">

and I can only look at the content when I select the whole text: 
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/39f33c33-3878-44dc-876d-0fd080a400e9">

With my fork it looks like:
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/180f0c3d-2936-49fe-91d6-8a0b6bf4ac47">

and the monitor page looks like:
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/0d65fed9-bb52-4076-8a79-e7dd9b9a4b5a">

The downside being that the colors looks more "faded" when using a darker theme:
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/fac98411-719e-4a5b-9891-6c09a68ff891">

compared to the current dashboard:
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/ff7efcf8-07c1-407f-9483-772353b1f730">

However, the "faded" one does honor the more darker white color of the scheme itself.

### Logs

None

### Environment

```txt
Client:
	Tag:         v1.8.0
	SHA:         5cc935f7
	Built:
	Go version:  go1.22.7
	OS/Arch:     darwin/arm64
Server:
	NODE:        192.168.2.110
	Tag:         v1.8.0
	SHA:         5cc935f7
	Built:
	Go version:  go1.22.7
	OS/Arch:     linux/amd64
	Enabled:     RBAC
```

**Comment by erinish** [comment]
I've run into this same issue and applied mostly the same patch for myself. Might be good to actually throw the PR together @Biodiscus else I feel like this is the kind of issue that doesn't get picked up because of how low priority it is. If there are some misgivings about color vibrancy it's probably easier to work out in a PR (though I'd argue that's the terminal's realm of responsibility).

**Comment by MaKaNu** [comment]
Have a similiar Issue with a dark theme:

![Image](https://github.com/user-attachments/assets/ef1d3e9f-fcb3-457b-9d72-40354c1e480c)

Terminal: alacritty 0.15.1 (0c405d53)

But I think it is based on the same issue: The color which the text is displaying is my white color. 
I think this is reasonable in my system to invert the colors for a dark theme.

**Comment by rothgar** [comment]
Is this still an issue? Happy to consider a PR that would fix this upstream.

**Comment by erinish** [comment]
I was beginning to put a PR together, but honestly I think this is actually not talosctl's problem to solve after thinking about it more. The crux of the issue is that there exist terminal themes where, for example, the BrightWhite ANSI color == the Background color. Famously, Solarized has this issue, which is what the screenshots in the OP are showing. 

I don't think there is a good solve for this scenario since the request for BrightWhite should not result in invisible elements on the background color in a reasonably designed theme. 

One could probably compare color elements against the background color and provide some contrast defaults in the case where they are actually the same color, but it's basically working around a pooly-designed theme and one would have to do this for all colors since you'd run into the same issue with say, BrightGreen if BrightGreen == the Background color in a given theme instead.

**Comment by rothgar** [comment]
I agree we can't fix that for all color combinations. Maybe if we know incompatible themes we could somehow manually override it with a known good theme or present an error to a user.

**Comment by Xnyle** [comment]
Had the same problem with Solarized and not only TC dashboard but also some other system monitors/tools.

I just wrapped this:
https://stackoverflow.com/questions/2507337/how-to-determine-a-terminals-background-color#7767891
around the call which changes background to black and reverses that after exit
```
blackterm(){
  oldstty=$(stty -g)

  # Such that no input is needed for the read command:
  stty raw -echo min 0 time 0

  #11 for background color
  printf "\033]11;?\033\\"

  # xterm needs the sleep
  sleep 0.1
  read -r original_background

  # Change background color for #fff000 (as an example)
  printf  "\033]11;#222222\033\\"
}

stdterm(){

  # Restore background-color.
  printf "\033$original_background\\"

  # Restore stty
  stty $oldstty
}
```
